rotkehlchen/chain/mixins/rpc_nodes.py

```diff
@@ -0,0 +1,374 @@
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+WEB3_NODE_TYPE = TypeVar('WEB3_NODE_TYPE', Web3, Client)
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class RPCNode(NamedTuple, Generic[WEB3_NODE_TYPE]):
+    """This represents an RPC node with its capabilities."""
+    rpc_client: WEB3_NODE_TYPE
+    is_pruned: bool
+    is_archive: bool
+
+
+class RPCManagerMixin(ABC, Generic[WEB3_NODE_TYPE]):
+    """
+    Mixin that provides logic for managing RPC nodes. It tracks active connections
+    and implements the core mechanisms for connecting to them.
+
+    This mixin is intended for chains that support RPCs, such as EVM-based chains
+    and Solana. It is not used directly, since each chain introduces small
+    differences in behavior. Instead, chain-specific mixins (e.g. `EVMRPCMixin`,
+    `SolanaRPCMixin`) extend this class and adjust parameters such as
+    `WEB3_NODE_TYPE` accordingly.
+    """
+    database: 'DBHandler'
+    greenlet_manager: GreenletManager
+    blockchain: SupportedBlockchain
+    chain_name: str
+    rpc_timeout: int
+
+    @staticmethod
+    def _connect_task_prefix(chain_name: str) -> str:
+        """Helper function to create the connection task greenlet name"""
+        return f'Attempt connection to {chain_name} node'
+
+    def __init__(self) -> None:
+        self.rpc_mapping: dict[NodeName, RPCNode[WEB3_NODE_TYPE]] = {}
+        # failed_to_connect_nodes keeps the nodes that we couldn't connect while
+        # doing remote queries so they aren't tried again if they get chosen. At the
+        # moment of writing this we don't remove entries from the set after some time.
+        # To force the app to retry a node a restart is needed.
+        self.failed_to_connect_nodes: set[str] = set()
+
+    def connected_to_any_web3(self) -> bool:
+        return len(self.rpc_mapping) != 0
+
+    def get_own_node_web3(self) -> WEB3_NODE_TYPE | None:
+        for node, rpc_node in self.rpc_mapping.items():
+            if node.owned:
+                return rpc_node.rpc_client
+        return None
+
+    def get_own_node_info(self) -> NodeName | None:
+        for node in self.rpc_mapping:
+            if node.owned:
+                return node
+        return None
+
+    def get_connected_nodes(self) -> list[NodeName]:
+        return list(self.rpc_mapping.keys())
+
+    def maybe_connect_to_nodes(self, when_tracked_accounts: bool) -> None:
+        """Start async connect to the saved nodes for the given blockchain if needed.
+
+        If `when_tracked_accounts` is True then it will connect when we have some
+        tracked accounts in the DB. Otherwise when we have none.
+
+        In ethereum case always connect to nodes. Needed for ENS resolution.
+        For other EVM chains we respect `when_tracked_accounts`.
+        """
+        if self.connected_to_any_web3() or self.greenlet_manager.has_task(self._connect_task_prefix(self.chain_name)):  # noqa: E501
```

Comment:
This function also needs renaming

```suggestion
        if self.connected_to_any_node() or self.greenlet_manager.has_task(self._connect_task_prefix(self.chain_name)):  # noqa: E501
```